tests/integration_tests/reaper.py

```diff
@@ -0,0 +1,194 @@
+"""Destroy instances in a background thread
+
+interface:
+
+start_reaper()                       - spawns reaper thread
+stop_reaper()                        - join thread and report leaked instances
+reap(instance: IntegrationInstance)  - queues instance for deletion
+
+start_reaper() / stop_reaper() - must be called only once
+"""
+
+import logging
+import queue
+import threading
+import warnings
+from typing import Final, List
+
+from tests.integration_tests.instances import IntegrationInstance
+
+LOG = logging.getLogger()
+
+
+class Reaper:
```

Comment:
Based on the immediate instantiation afterwards I'm assuming this class is intended to be used as a singleton? If so, I think we should make the class name `_Reaper` so that it's more obvious.

But there's two instances. In conftest.py, you instantiate a new instance, but then in instances.py you reference the instance created in this file.

Also, the module docstring is out of date and I think function names like `reaper_start()` and `reaper_stop()` can be updated to just `start()` and `stop()` if they live inside the class.

Reply:
> I'm assuming this class is intended to be used as a singleton

Yes

> But there's two instances.

Ahhhhhh, nice catch